parser-typechecker/src/Unison/Codebase/Path.hs

```diff
@@ -288,16 +294,15 @@ uncons = Lens.uncons
 -- todo: fromName needs to be a little more complicated if we want to allow
 --       identifiers called Function.(.)
 fromName :: Name -> Path
-fromName = fromList . fmap NameSegment . Text.splitOn "." . Name.toText
+fromName = fromList . Name.segments
 
 fromName' :: Name -> Path'
-fromName' n = case first of
-  [NameSegment ""] -> Path' . Left . Absolute . Path $ Seq.drop 1 seq
-  _    -> Path' . Right $ Relative path
+fromName' n = case take 1 (Name.toString n) of
+  "." -> Path' . Left . Absolute $ Path seq
+  _   -> Path' . Right $ Relative path
  where
-  path  = fromName n
-  seq   = toSeq path
-  first = Seq.take 1 seq
+  path = fromName n
+  seq  = toSeq path
```

Comment:
Since `fromName` works differently now, we can't rely on the hack that the first segment is empty to detect the root.

Reply:
Not sure I quite got this — is there anything else we need to know about how `fromName` works differently now?

Reply:
The way it works differently is that instead of `...` giving the list `["","","",""]` (the empty strings between the dots), it gives `[".","."]` (the two namespaces named `"."` separated by a dot).